detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/Project/ProjectReferenceResolver.cs

```diff
@@ -32,8 +43,19 @@ public DependencyResult Process()
                 List<NugetDependency> deps = new List<NugetDependency>();
                 foreach (ProjectItem reference in proj.GetItemsIgnoringCondition("PackageReference"))
                 {
+                    bool containsPkg = Packages != null && Packages.Any(pkg => pkg.Name.Equals(reference.EvaluatedInclude));
                     var versionMetaData = reference.Metadata.Where(meta => meta.Name == "Version").FirstOrDefault();
-                    if (versionMetaData != null)
+                    var versionOverrideMetaData = reference.Metadata.Where(meta => meta.Name == "VersionOverride").FirstOrDefault();
+                    if (containsPkg && versionOverrideMetaData != null && CheckVersionOverride)
```

Comment:
From https://learn.microsoft.com/en-us/nuget/consume-packages/central-package-management:

> When this feature is disabled, specifying a VersionOverride on any <PackageReference /> item will result in an error at restore time indicating that the feature is disabled.

I wonder if we should halt parsing if overrides are disabled but an override is specified. What do you think?

Reply:
Sounds good, yeah I will halt parsing and report an error in that case.